File: mixed_media/richest_customer_wealth/wealth.hs
import System.Random (randomRIO)
import Control.Monad (replicateM)

maximumWealth :: [[Int]] -> Int
maximumWealth = maximum . map sum

generateMatrix :: Int -> Int -> Int -> Int -> IO [[Int]]
generateMatrix m n minV maxV = replicateM m $ replicateM n $ randomRIO (minV, maxV)

main :: IO ()
main = do
    let (m, n, minV, maxV) = (3, 3, -10000, 100000) -- Rows, Columns, Min, Max
    accounts <- generateMatrix m n minV maxV
    mapM_ (putStrLn . unwords . map show) accounts -- Print matrix
    print $ maximumWealth accounts

File: mixed_media/richest_customer_wealth/wealth.rs
pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
    accounts.iter()
            .map(|x| x.iter().sum()) // Row sums
            .max()                  // max row sum
            .unwrap()              // unwrap the results
}

fn main() {
    let accounts = vec![
        vec![1, 2, 3, 4],
        vec![5, 6, 7, 8],
        vec![9, 10, 11, 12]
    ];
  
    let max_wealth = maximum_wealth(accounts);
    println!("Maximum wealth: {}", max_wealth);
}
